cmd/bastion/aws/destroy.go

```diff
@@ -0,0 +1,202 @@
+package aws
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	hyperv1 "github.com/openshift/hypershift/api/v1alpha1"
+	awsutil "github.com/openshift/hypershift/cmd/infra/aws/util"
+	"github.com/openshift/hypershift/cmd/util"
+)
+
+type destroyBastionOpts struct {
+	Namespace          string
+	Name               string
+	InfraID            string
+	Region             string
+	AWSCredentialsFile string
+}
+
+func NewDestroyCommand() *cobra.Command {
+	opts := destroyBastionOpts{
+		Namespace: "clusters",
+	}
+	cmd := &cobra.Command{
+		Use:          "aws",
+		Short:        "Destroys AWS bastion instance",
+		SilenceUsage: true,
+	}
+
+	cmd.Flags().StringVar(&opts.Namespace, "namespace", opts.Namespace, "The namespace of the hostedcluster")
+	cmd.Flags().StringVar(&opts.Name, "name", opts.Name, "The name of the hostedcluster")
+	cmd.Flags().StringVar(&opts.InfraID, "infra-id", opts.InfraID, "The infra ID to use for creating the bastion")
+	cmd.Flags().StringVar(&opts.Region, "region", opts.Region, "The region to use for creating the bastion")
+	cmd.Flags().StringVar(&opts.AWSCredentialsFile, "aws-creds", opts.AWSCredentialsFile, "File with AWS credentials")
+
+	cmd.MarkFlagRequired("aws-creds")
+	cmd.MarkFlagFilename("aws-creds")
+
+	cmd.Run = func(cmd *cobra.Command, args []string) {
+		if err := opts.Validate(); err != nil {
+			log.Error(err, "Invalid arguments")
+			cmd.Usage()
+			return
+		}
+		ctx, cancel := context.WithCancel(context.Background())
+		sigs := make(chan os.Signal, 1)
+		signal.Notify(sigs, syscall.SIGINT)
+		go func() {
+			<-sigs
+			cancel()
+		}()
+
+		if err := opts.Run(ctx); err != nil {
+			log.Error(err, "Failed to create bastion")
+			os.Exit(1)
+		} else {
+			log.Info("Successfully destroyed bastion")
+		}
+	}
+
+	return cmd
+}
+
+func (o *destroyBastionOpts) Validate() error {
+	if len(o.Name) > 0 {
+		if len(o.Namespace) == 0 {
+			return fmt.Errorf("a namespace must be specified if specifying a hosted cluster name")
+		}
+		if len(o.InfraID) > 0 || len(o.Region) > 0 {
+			return fmt.Errorf("infra id and region cannot be specified when specifying a hosted cluster name")
+		}
+	} else {
+		if len(o.InfraID) == 0 || len(o.Region) == 0 {
+			return fmt.Errorf("infra id and region must be specified when not specifying a hosted cluster name")
+		}
+	}
+	return nil
+}
+
+func (o *destroyBastionOpts) Run(ctx context.Context) error {
+
+	var infraID, region string
+
+	if len(o.Name) > 0 {
```

Comment:
I feel like we should make the whole "Default region and infraID if unset from hostedcluster in api", we have it like five times now?

Reply:
The issue is that in other commands like `create cluster` and `destroy cluster` we need more than just those 2 like domain